### PR TITLE
Feature v0.13.0 mq register

### PR DESCRIPTION
**Pull request by mattkasun**
new function to register with server and obtain CA cert

### CODE REVIEW

File: compose/docker-compose.contained.yml [hunk 1]
```diff
@@ -36,7 +37,7 @@ services:
       MQ_HOST: "mq"
       HOST_NETWORK: "off"
       MANAGE_IPTABLES: "on"
-      PORT_FORWARD_SERVICES: "mq"
+      PORT_FORWARD_SERVICES: ""
```
Comment (0xdcarns) [inline compose/docker-compose.contained.yml:40]: should just remove this variable

File: netclient/command/commands.go [hunk 2]
```diff
@@ -154,3 +154,7 @@ func Daemon() error {
 	err := functions.Daemon()
 	return err
 }
+
+func Register(cfg *config.ClientConfig, key string) error {
```
Comment (0xdcarns) [inline netclient/command/commands.go:158]: remove

File: netclient/functions/register.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		if err := RegisterWithServer(private, cfg); err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
+	return JoinNetwork(cfg, key, false)
+}
+
+func RegisterWithServer(private *ed25519.PrivateKey, cfg *config.ClientConfig) error {
```
Comment (0xdcarns) [inline netclient/functions/register.go:53]: missing comment

File: tls/tls.go [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+	"math/big"
+	"os"
+	"time"
+
+	"filippo.io/edwards25519"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+// certificate validity in days
+const CERTIFICATE_VALIDITY = 365
```
Comment (0xdcarns) [inline tls/tls.go:21]: malformed comment should be CERTIFICATE_VALIDITY - blah blah

File: tls/tls.go [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+	if n.point == nil {
+		return ed25519.PrivateKey{}, errors.New("nil point")
+	}
+	if len(n.point.Bytes()) != ed25519.SeedSize {
+		return ed25519.PrivateKey{}, errors.New("incorrect seed size")
+	}
+	return ed25519.NewKeyFromSeed(n.point.Bytes()), nil
+}
+
+// Curve25519PrivateKey returns the private key in Montogomery form used for ECDH.
```
Comment (0xdcarns) [inline tls/tls.go:48]: should be Key.Curve25519PrivateKey -

File: tls/tls.go [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+// NewKey generates a new key.
+func NewKey() *Key {
+	seed := make([]byte, 64)
+	rand.Reader.Read(seed)
+	s, _ := (&edwards25519.Scalar{}).SetUniformBytes(seed)
+	return &Key{(&edwards25519.Point{}).ScalarBaseMult(s)}
+}
+
+// Ed25519PrivateKey returns the private key in Edwards form used for EdDSA.
+func (n *Key) Ed25519PrivateKey() (ed25519.PrivateKey, error) {
```
Comment (0xdcarns) [inline tls/tls.go:38]: should be Key.Ed25519PrivateKey -

File: tls/tls.go [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+		return wgtypes.Key{}, errors.New("nil point")
+	}
+	if len(n.point.Bytes()) != ed25519.SeedSize {
+		return wgtypes.Key{}, errors.New("incorrect seed size")
+	}
+	return wgtypes.ParseKey(base64.StdEncoding.EncodeToString(n.point.BytesMontgomery()))
+}
+
+// Save : saves the private key to path.
+func (n *Key) Save(path string) error {
```
Comment (0xdcarns) [inline tls/tls.go:60]: Key.Save comment

File: tls/tls.go [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	f.Write(n.point.Bytes())
+	return nil
+}
+
+// Reads the private key from path.
+func ReadFrom(path string) (*Key, error) {
```
Comment (0xdcarns) [inline tls/tls.go:71]: ReadFrom reads the...

File: tls/tls.go [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+	"filippo.io/edwards25519"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+// certificate validity in days
+const CERTIFICATE_VALIDITY = 365
+
+type (
+	Key struct {
```
Comment (0xdcarns) [inline tls/tls.go:24]: // Key is the struct for an edwards representation point

Comment (mattkasun) [reply tls/tls.go:24]: added comment

**Pull request merged**